cibuildwheel/ios.py

```diff
@@ -0,0 +1,593 @@
+        venv_path,
+        dependency_constraint_flags,
+        use_uv=False,
+    )
+
+    # Convert the macOS virtual environment into an iOS virtual environment
+    # using the cross-platform conversion script in the iOS distribution.
+
+    # target_python is the path to the Python binary;
+    # determine the root of the XCframework slice that is being used.
+    slice_path = target_python.parent.parent
+    call(
+        "python",
+        str(slice_path / f"platform-config/{multiarch}/make_cross_venv.py"),
+        str(venv_path),
+        env=env,
+        cwd=venv_path,
+    )
+
+    # When running on macOS, it's easy for the build environment to leak into
+    # the target environment, especially when building for ARM64 (because the
+    # build architecture is the same as the target architecture). The primary
+    # culprit for this is Homebrew libraries leaking in as dependencies for iOS
+    # libraries.
+    #
+    # To prevent problems, set the PATH to isolate the build environment from
+    # sources that could introduce incompatible binaries.
+    env["PATH"] = os.pathsep.join(
+        [
+            # The target python's binary directory
+            str(target_python.parent),
+            # The cross-platform environments binary directory
+            str(venv_path / "bin"),
+            # Cargo's binary directory (to allow for Rust compilation)
+            str(Path.home() / ".cargo" / "bin"),
+            # The bare minimum Apple system paths.
+            "/usr/bin",
+            "/bin",
+            "/usr/sbin",
+            "/sbin",
+            "/Library/Apple/usr/bin",
+        ]
+    )
+    # Also unset DYLD_LIBRARY_PATH to ensure that no macOS libraries will be
+    # found and linked.
+    env.pop("DYLD_LIBRARY_PATH", None)
+
+    return env
+
+
+def setup_python(
+    tmp: Path,
+    python_configuration: PythonConfiguration,
+    dependency_constraint_flags: Sequence[PathOrStr],
+    environment: ParsedEnvironment,
+    build_frontend: BuildFrontendName,
+) -> tuple[Path, dict[str, str]]:
+    if build_frontend == "build[uv]":
+        msg = "uv doesn't support iOS"
+        raise ValueError(msg)
+
+    # An iOS environment requires 2 python installs - one for the build machine
+    # (macOS), and one for the target (iOS). We'll only ever interact with the
+    # *target* python, but the build Python needs to exist to act as the base
+    # for a cross venv.
+    tmp.mkdir()
+    implementation_id = python_configuration.identifier.split("-")[0]
+    log.step(f"Installing Build Python {implementation_id}...")
+    if implementation_id.startswith("cp"):
+        free_threading = "t-ios" in python_configuration.identifier
+        build_python = install_build_cpython(
+            tmp,
+            python_configuration.version,
+            python_configuration.build_url,
+            free_threading,
+        )
+    else:
+        msg = "Unknown Python implementation"
+        raise ValueError(msg)
```

Comment:
```suggestion
        msg = f"Unknown Python implementation: {implementation_id}"
        raise errors.FatalError(msg)
```
or we shouldn't be here at all by design in which case an `assert implementation_id.startswith("cp")` might be more appropriate ?